LBHackney-IT/lbh-fss-public-frontend
# Map functionality

### Setup
Following steps added to README, please apply:
1.  Copy and rename the `Mapbox.example.js` as `Mapbox.js` in `src/helpers/Mapbox` and replace the token string with your Mapbox token.
2. Copy and rename the `example.npmrc` as `.npmrc` in the project root and replace the authToken with your Font Awesome pro token.
3. Run `npm install` from project root to install packages and dependencies.
4. FYI new test data has been added to the mockServices.json fine to include 12 markers within the hackney area

Credentials can be found in 1pw `Mapbox login - nudge - hackney` and `Font awesome pro token - hackney`

### Test
- Note: there is 1 map component (and 1 placeholder, used for the home and setting pages), so any issues found across multiple pages should just be a single issue, but log as necessary
- Ensure placeholder map is correctly displayed on the home and settings page (set postcode, select categories / demographics) - map does not include markers and is only visible for desktop
- Ensure map is displayed on all other pages
  - with pins correctly associated with different category colours
  - multiple location markers to a service displayed correctly
- View as list / map functionality and state stored correctly when viewing lists of services on a smaller screen (this feature should not be shown on desktop) - the state should be carried through when navigating through pages
- Clicking on a marker will display a popup with the card details. Clicking the card will direct the user to the service detail
- On the service detail markers do not display a popup and should only display location markers associated to this service

### Tasks
https://app.clickup.com/t/9e5u58
https://app.clickup.com/t/82pe98

### To implement later
- Investigate the popup display on smaller devices (in the design it shows as fixed to the bottom of page)
- Home postcode marker will be provided by API data
- Distance will be provided by API data
